test/unit/streaming-client.spec.ts

```diff
@@ -249,6 +253,7 @@ describe('streaming-client', () => {
 
             // Assert
             expect(() => { duplex.write("message"); }).toThrow();
+            expect(duplex.writable).toBe(false);
```

Comment:
test currently fails on this because the duplex doesn't emit the `end` event until the input stream drains the final EOS message and the stream isn't marked as unwritable until the `end` event fires. Working on finding a way to make the client's end function wait until either the `end` event or the `drain` event are fired before returning